### PR TITLE
Release 2018.10 notes

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This is the draft PR for the 2018.10 release note (follow up of @cladmi's idea to do it in a PR :)

I invite maintainers to push directly to my branch to document major changes that are not yet listed.
Developers can comment here to add their work.

As a reference, I tried to list all merged PRs in master and closed issues for this release in https://github.com/RIOT-OS/RIOT/milestone/24?closed=1.
It will be again updated with time.

### Issues/PRs references
Release 2018.10

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,228 @@
...
+#4954: chronos: compiling with -O0 breaks
+#4866: periph: GPIO drivers are not thread tafe
+#5009: RIOT is saw-toothing in energy consumption (even when idling)
+#5128: some use of asm keyword might be missing volatile
+#5486: at86rf2xx: lost interrupts
+#5774: cpu: cortexm_common: context switching code breaks when compiling with
+       LTO
+#5848: arduino: Race condition in sys/arduino/Makefile.include
+#6018: nRF52: gnrc 6lowpan ble memory leak
+#6437: periph/spi: Leftovers from SPI rework in #4780
```
Comment (aabadie) [inline release-notes.txt]: This issue can be removed since it is closed now. Maybe it should just be moved to the solved issues.

Comment (jia200x) [reply release-notes.txt]: ok! I will re-check the list now

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,228 @@
...
+#6567: periph/spi: Switching between CPOL=0,1 problems on Kinetis with software
+       CS
+#6650: GPIO Port shift in cc2538 is unconsistent
+#6838: Issuing a reboot on ATmega328p makes the board loop-reboot
+#6874: SAMD21: possible CMSIS bug
+#7020: isr_rfcoreerrors while pinging between CC2538DKs
+#7667: sam0 flashpage_write issues
+#7846: stm32f1: I2C read functions return bogus values with unconnected devices
```
Comment (aabadie) [inline release-notes.txt]: This was fixed/closed by the I2C refactoring

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,228 @@
...
+#5776: make: Predefining CFLAGS are parsed weirdly
+#5863: OSX+SAMR21-xpro: shell cannot handle command inputs larger than 64 chars
+#6120: Windows AVR Mega development makefile Error
+#7220: sys/fmt: Missing tests for fmt_float,fmt_lpad
+#7877: SPI connection can't read 16bit size register
+#8107: crypto/ccm: bugs in the implementation of CCM mode
+#9548: Many i2c bugs found with the i2c api rework but did not reach the release
+#9709: examples: failed assertion in dtls-echo example
+
+Fixed Issues from the last release (2018.04)
```
Comment (aabadie) [inline release-notes.txt]: Should be 2018.07. Is this just a copy paste from the previous release ?

**Comment by jia200x** [comment]
thank you so much @aabadie!!

File: release-notes.txt [hunk 4]
```diff
@@ -38,43 +38,61 @@ New features and changes
...
-    TBD
+    + drivers/vcnl40x10: added initial support
+    * drivers/periph_i2c: rework API
+    * drivers/*: unify driver initialization parameters scheme
+    * drivers/ata8520e: rework internal implementation
+
 Build System / Tooling
 ----------------------
-    TBD
+    * pyterm: handle ctrl+d shortcut nicely
```
Comment (jia200x) [inline release-notes.txt]: this was not backported in the end :(. It changes an API

File: release-notes.txt (hunk 3 above)
Comment (jia200x) [reply release-notes.txt]: yes, I should had marked it as WIP

File: release-notes.txt (hunk 4 above)
Comment (aabadie) [reply release-notes.txt]: This point is related to #8970, which was in master before the feature freeze, so there was no need for backporting.

Comment (jia200x) [reply release-notes.txt]: ah! sorry, I though it was #10105

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,259 @@
...
+      Hashes)
+    * Drivers with saul support are now also part of the drivers_saul group
+
+Core
+----
+    TBD
+
+System libraries
+----------------
+
```
Comment (miri64) [inline release-notes.txt]:
```
     * tiny-asn1: bump version
```

File: release-notes.txt [hunk 6]
```diff
@@ -1,3 +1,259 @@
...
+    + add support for esp32-mh-et-live-minikit
+    + add support for esp32-olimex-evb
+    + add support for esp32-wemos-lolin-d32-pro
+    + add support for esp32-wroom-32
+    + add support for esp32-wrover-kit
+    + add support for samr30-xpro
+    + add support for ublox-c030-u201
+    + add i2c configuration for nucleo-f070, nucleo-f091rc and nucleo-f722ze
+    + add second i2c configuration for b-l475e-iot01a
+
```
Comment (miri64) [inline release-notes.txt]:
```
    + add PWM support for NRF52
```

**Comment by jia200x** [comment]
btw [this gist](https://gist.github.com/jia200x/03a2fd42ec2195fdedbc0bb7f634ce2a) contains all PRs between 2018.07-RC2 (Release) and 2018.10-RC2. I tagged them with their "Type: " label in the beggining, so it's easy to filter them using grep ;)

**Comment by cladmi** [comment]
@jia200x Why not just using https://github.com/RIOT-OS/RIOT/milestone/24 ?

**Comment by jia200x** [comment]
> @jia200x Why not just using https://github.com/RIOT-OS/RIOT/milestone/24 ?

Because I can copy them directly from the grep'd/sed'd file to the release-notes :)

**Comment by cladmi** [comment]
I found that the milestone was outdated, I updated my PRs to be listed in the milestone at least.

**Comment by jia200x** [comment]
> I found that the milestone was outdated, I updated my PRs to be listed in the milestone at least.

Thank you! Yeah, that's also why I got it from the commits. I was afraid it might be not synced properly in some cases

**Comment by cladmi** [comment]
I added my contributions after the others in the list.

>  I was afraid it might be not synced properly in some cases

It is better if it can be updated for the release ;)
To update it during the previous release I used this page https://github.com/RIOT-OS/RIOT/pulls?q=is%3Amerged+no%3Amilestone

I also used it this time to find my unasigned PRs.

File: release-notes.txt [hunk 7]
```diff
@@ -1,3 +1,270 @@
...
+    * cortexm_common/ldscript: simplify generation of firmwares in section of
+      the ROM
+    * cpu/stm32_common: fix source selection declared as module dependencies
+
+Device Drivers
+--------------
+    + drivers/vcnl40x10: added initial support
+    * drivers/periph_i2c: rework API
+    * drivers/*: unify driver initialization parameters scheme
+    * drivers/ata8520e: rework internal implementation
```
Comment (danpetry) [inline release-notes.txt]: tsl4531x and ds18 device drivers got added

File: release-notes.txt [hunk 8]
```diff
@@ -1,3 +1,270 @@
...
+--------------------
+
+#5103: xtimer: weird behavior of tests/xtimer_drift
+#5338: xtimer: xtimer_now() is not ISR safe
+#6052: tests: xtimer_drift_gets_stuck
+#7114: xtimer: adds items to the wrong list if the timer overflows between
+       _xtimer_now() and irq_disable()
+#7347: xtimer_usleep stuck for small values
+#8251: telosb: timer config wrong when running o a tmote sky
+#9052: misc issues with tests/trickle
```
Comment (danpetry) [inline release-notes.txt]: #10073: timer_usleep wrong for small delay times (https://github.com/RIOT-OS/RIOT/issues/10073)

File: release-notes.txt [hunk 9]
```diff
@@ -1,3 +1,401 @@
...
+    + libb2: add support (BLAKE2)
+    + libhydrogen: add support
+    + lora_serialization: LoRa Serialization format support.
+    * tiny-asn1: bump version
+    * openthread: rework of FTD and MTD support
+    * semtech-loramac: bump to version 4.4.1
+    * semtech-loramac: use periph_eeprom to store lorawan configuration
+    * semtech-loramac: add support for RU864 ISM band
+    * semtech-loramac: lorawan support is not considered experimental anymore
+    * semtech-loramac: reduce timeout in timer
```
Comment (jia200x) [inline release-notes.txt]: @aabadie I will remove this was a revert of a change introduced by the `semtech_loramac` bump during the release ;)

File: release-notes.txt [hunk 10]
```diff
@@ -80,6 +84,7 @@ Boards
     + add support for ublox-c030-u201
     + add i2c configuration for nucleo-f070, nucleo-f091rc and nucleo-f722ze
     + add second i2c configuration for b-l475e-iot01a
+    + add PWM support for NRF52
```
Comment (jia200x) [inline release-notes.txt]: @aabadie I moved this one to CPU

**Comment by cladmi** [comment]
I added `https://github.com/RIOT-OS/RIOT/pull/10377` to the release notes as it should still go in.

**Comment by cladmi** [comment]
Also added https://github.com/RIOT-OS/RIOT/pull/10374 that I just merged.

File: release-notes.txt [hunk 11]
```diff
@@ -1,3 +1,390 @@
...
+    * docker: export RIOT_CI_BUILD_to docker
+    * docker: handle building in git worktree
+    * edbg.inc.mk: allow flashing with an offset in rom without erasing all ROM
+    * make: add APPDIR to info-build target
+    * make: provide support for listing supported and blacklisting toolchains
+    * make: unify app folder search (examples/*, tests/*, ...)
+    * makefiles/buildtests.inc.mk: fix 'clean-intermediates' not cleaning
+    * makefiles/docker.inc.mk: handle building in git worktree
+    * openocd.sh: allow flashing binary files without configuration
+    * pkg.mk: do not use user identidy when applying patches
```
Comment (MrKevinWeiss) [inline release-notes.txt]: identidy -> identity

File: release-notes.txt [hunk 12]
```diff
@@ -1,3 +1,390 @@
...
+============================================
+
+#9573: pkg/lwip: assert on samr21-xpro
+#9584: drivers/cc2420: add netopt NETOPT_MAX_PACKET_SIZE
+#6437: periph/spi: Leftovers from SPI rework in #4780
+#6650: GPIO Port shift in cc2538 is unconsistent
+#6836: Issuing a reboot on ATmega328p makes the board loop-reboot
+#7846: stm32f1: I2C read functions return bogus values with unconnected devices
+#9616: at86rf2xx: can not correctly set channel after `NETOPT_STATE_RESET`
+#9548: Many i2c bugs found with the i2c api rework but did not reach the release
```
Comment (MrKevinWeiss) [inline release-notes.txt]: Should this be in both known and fixed issue (I guess certain aspects were fixed and some remain?

Comment (MrKevinWeiss) [reply release-notes.txt]: specifically this line: 
#9548: Many i2c bugs found with the i2c api rework but did not reach the release

**Comment by jia200x** [comment]
wrote a script for last checks using the Github API. It turns out some of the fixed issues were still in the Know Issues part. So, I removed these stuff in the last commit ;)

**Comment by jia200x** [comment]
(I will release these helpers when I find some time)

File: release-notes.txt (hunk 2 above)
Comment (jia200x) [reply release-notes.txt]: done!

File: release-notes.txt (hunk 3 above)
Comment (jia200x) [reply release-notes.txt]: done!

**Comment by jia200x** [comment]
@aabadie are you ok with the changes?

**Comment by miri64** [comment]
> It turns out some of the fixed issues were still in the Know Issues part. So, I removed these stuff in the last commit ;)

They should be moved to fixed issues then ;-)

**Comment by jia200x** [comment]
> They should be moved to fixed issues then ;-)

They are already there, they were duplicated :)

**Comment by miri64** [comment]
Ah okay. The commit did (of course) did not show that, so I did not know.

**Comment by jia200x** [comment]
@RIOT-OS/maintainers last call :) 

If the document is ok please ACK :)

File: release-notes.txt [hunk 13]
```diff
@@ -1,3 +1,384 @@
...
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release:
+===================
+
+The 2018.10 release includes new features, like CoAP Resource Directory and 
+Block2, ESP32 support, rework of I2C interface,  EEPROM registry,
```
[the remark this answers is not in the document]
Comment (jia200x) [reply release-notes.txt]: you are right!

File: release-notes.txt [hunk 14]
```diff
@@ -1,3 +1,384 @@
...
+Full and Minimal Thread Devices support in OpenThread and compliance of
+LoRaWAN (ABP counters stored in EEPROM). Several boards were introduced.
```
[the remark this answers is not in the document]
Comment (jia200x) [reply release-notes.txt]: sure!

**Comment by jia200x** [comment]
@aabadie done. I will squash now

**Pull request merged**